tools/stress-cluster/services/Stress.Watcher/src/JobEventHandler.cs

```diff
@@ -0,0 +1,237 @@
+                                    resourceVersion = null;
+                                }
+                            }
+                            tcs.TrySetException(err);
+                            throw err;
+                        },
+                        () => {
+                            Logger.Warning("Job watch has closed.");
+                            tcs.TrySetResult();
+                        }
+                    );
+                    using var registration = cancellationToken.Register(watcher.Dispose);
+                    await tcs.Task;
+                }
+                catch (Exception ex)
+                {
+                    Log.Error(ex, "Error with job watch stream.");
+                    await Task.Delay(1000, cancellationToken);
+                }
+            }
+        }
+
+        public void HandleJobEvent(WatchEventType type, V1Job job)
+        {
+            using (LogContext.PushProperty("namespace", job.Namespace()))
+            using (LogContext.PushProperty("job", job.Name()))
+            {
+                DeleteResources(job).ContinueWith(t =>
+                {
+                    if (t.Exception != null)
+                    {
+                        Logger.Error(t.Exception, "Error deleting resources.");
+                    }
+                });
+            }
+        }
+
+        public async Task DeleteResources(V1Job job)
+        {
+            if (!ShouldDeleteResources(job))
+            {
+                Logger.Debug($"Skipping resource deletion.");
+                return;
+            }
+
+            var rgName = GetResourceGroupName(job);
+
+            if (string.IsNullOrEmpty(rgName))
+            {
+                return;
+            }
+
+            Subscription subscription = ARMClient.DefaultSubscription;
+
+            ResourceGroup resourceGroup;
+            try {
+                resourceGroup = await subscription.GetResourceGroups().GetAsync(rgName);
+            } catch (Exception) {
+                Logger.Error($"Failed to get resource group '{rgName}' using subsription id '{subscription.Id}'");
+                return;
+            }
+
+            Logger.Information($"Deleting resources for group {rgName}");
+            await resourceGroup.DeleteAsync();
+            Logger.Information($"Deleted resources for group {rgName}");
+        }
+
+        public bool ShouldDeleteResources(V1Job job)
+        {
+            if (!string.IsNullOrEmpty(Namespace) && Namespace != job.Namespace())
+            {
+                return false;
+            }
+
+            var initContainers = job.Spec?.Template?.Spec?.InitContainers;
+            if (initContainers == null || initContainers.Count() == 0) {
+                return false;
+            }
+
+            var deployContainers = initContainers.Where(c => c.Name == "init-azure-deployer");
+            if (deployContainers.Count() == 0)
+            {
+                return false;
+            }
```

Comment:
might be better if we add in the reasons why delete resources are skipped? for easier debug and transparency of the watcher. Would be beneficial for the "skip deleting my job's resource" that Richard wanted as well.

Reply:
Yep, makes sense. I'll update.